### PR TITLE
chore(crds) upgrade to apiextensions.k8s.io/v1

### LINKED ISSUE #888: Deprecated `apiextensions.k8s.io/v1beta1` `CustomResourceDefinition`
### Summary

Migrate `CustomResourceDefinition` to `apiextensions.k8s.io/v1` as `apiextensions.k8s.io/v1beta1` is deprecated

```
`CustomResourceDefinition` represents a resource that should be exposed on the API server. Its name MUST
be in the format .spec.name>..spec.group>.
Deprecated in v1.16, planned for removal in v1.19. Use apiextensions.k8s.io/v1 CustomResourceDefinition instead.
```

### Steps To Reproduce

1. Check https://github.com/Kong/kuma/tree/master/pkg/plugins/resources/k8s/native/config/crd/bases

### Additional Details & Logs

- Version
- Error logs
- Configuration
- Platform and Operating System

### LINKED ISSUE #1687: Upgrade Webhook Resources to admissionregistration.k8s.io/v1
### Summary

As noted in https://github.com/kumahq/kuma/issues/1680, the [`ValidatingWebhookConfiguration`](https://github.com/kumahq/kuma/blob/5785ce35cfa99083fff09a1a645f11129e2a0cfb/deployments/charts/kuma/templates/cp-webhooks-and-secrets.yaml#L122-L204) and [`MutatingWebhookConfiguration`](https://github.com/kumahq/kuma/blob/5785ce35cfa99083fff09a1a645f11129e2a0cfb/deployments/charts/kuma/templates/cp-webhooks-and-secrets.yaml#L48-L120) use the deprecated `admissionregistration.k8s.io/v1beta1` API, which will be removed in k8s v1.22. 

For migration, see: https://kubernetes.io/docs/reference/using-api/deprecation-guide/#webhook-resources-v122

* All existing persisted objects are accessible via the new APIs
* Notable changes:
   ```
    webhooks[*].failurePolicy default changed from Ignore to Fail for v1
    webhooks[*].matchPolicy default changed from Exact to Equivalent for v1
    webhooks[*].timeoutSeconds default changed from 30s to 10s for v1
    webhooks[*].sideEffects default value is removed, and the field made required, and only None and NoneOnDryRun are permitted for v1
    webhooks[*].admissionReviewVersions default value is removed and the field made required for v1 (supported versions for AdmissionReview are v1 and v1beta1)
    webhooks[*].name must be unique in the list for objects created via admissionregistration.k8s.io/v1
   ```
### Additional Details & Logs

- Version `kuma v1.1.1`

### PR DESCRIPTION
**Pull request by austince**
### Summary

Upgrades our Custom Resource Definitions to `apiextensions.k8s.io/v1`, as `apiextensions.k8s.io/v1beta1` is deprecated. This drops support for k8s version < 1.16. 

Migration notes:
* https://github.com/kubernetes/kubernetes/pull/79604
* https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#validation
* https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#specifying-a-structural-schema

### Full changelog

* Upgrade all chart CRDs to apiextensions.k8s.io/v1, which involved just dropping the native k8s types from the openapi spec.

### Issues resolved

Fix #888
Fix #1687

### Documentation

- I couldn't find a place where we note the minimum supported k8s version -- have I missed that somewhere/ should we add it if not?

### CODE REVIEW

**Comment by nickolaev** [comment]
Oh, removing 5,000 unneeded lines, great! You will probably have to regenerate the kumactl embedded helm templates. Also the golden files for the various `kumactl install control-plane` tests.
I would keep this one for post 1.0.0 as you suggest. Dropping `<1.16` is not a trivial thing to state just like that. I see k8s 1.20 release is scheduled for early Dec'20, https://www.kubernetes.dev/resources/release/ Maybe that is a good milestone for this one to get in.

In any case, thanks - awesome work!

**Comment by austince** [comment]
> Oh, removing 5,000 unneeded lines, great! You will probably have to regenerate the kumactl embedded helm templates. Also the golden files for the various `kumactl install control-plane` tests.

10k lines now :astonished:  

> I would keep this one for post 1.0.0 as you suggest. Dropping `<1.16` is not a trivial thing to state just like that. I see k8s 1.20 release is scheduled for early Dec'20, https://www.kubernetes.dev/resources/release/ Maybe that is a good milestone for this one to get in.

Yup, this sounds good to me - would like to update the webhooks as well which will also only support 1.16+

File: pkg/plugins/resources/k8s/native/config/crd/bases/kuma.io_circuitbreakers.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 
 ---
-apiVersion: apiextensions.k8s.io/v1beta1
+apiVersion: apiextensions.k8s.io/v1
```
Comment (austince) [inline pkg/plugins/resources/k8s/native/config/crd/bases/kuma.io_circuitbreakers.yaml:3]: Can we remove this entire `pkg/plugins/resources/k8s/native` directory as well? I think it's confusing to have a kubebuilder operator that is not used.

File: pkg/plugins/runtime/k8s/controllers/namespace_controller.go [hunk 2]
```diff
@@ -3,7 +3,7 @@ package controllers
 import (
 	"context"
 
-	"k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
+	v1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
```
Comment (austince) [inline pkg/plugins/runtime/k8s/controllers/namespace_controller.go:6]: nit, could remove the `v1` alias or change it to `apiextensionsv1`?

**Review by austince — COMMENTED** [review]
Looks good @nickolaev, just a nit and a question about further removals.

File: pkg/plugins/resources/k8s/native/config/crd/bases/kuma.io_circuitbreakers.yaml (hunk 1 above)
Comment (nickolaev) [reply pkg/plugins/resources/k8s/native/config/crd/bases/kuma.io_circuitbreakers.yaml:3]: Tried it, but not that easy though. Maybe we have to do it in a dedicated PR.

Comment (austince) [reply pkg/plugins/resources/k8s/native/config/crd/bases/kuma.io_circuitbreakers.yaml:3]: Sad, thanks for trying.

**Comment by austince** [comment]
Changes to the webhooks look good as well

**Comment by nickolaev** [comment]
> Changes to the webhooks look good as well

Reverted those, not working without substantial code changes. We still have to do this but in a separate PR. Let's converge and merge this one.

File: pkg/plugins/runtime/k8s/controllers/service_controller.go [hunk 3]
```diff
@@ -4,7 +4,7 @@ import (
 	"context"
 	"fmt"
 
-	"k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
+	v1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
```
Comment (bartsmykla) [inline pkg/plugins/runtime/k8s/controllers/service_controller.go:7]: in other parts of the code it's called `apiextensionsv1`, do you think we could make it consistent?

**Review by bartsmykla — APPROVED** [review]
Just a nit with an import name. I would prefer to have the CI/CD changes in separate PR, but I don't have strong opinion.

**Comment by nickolaev** [comment]
> Just a nit with an import name. I would prefer to have the CI/CD changes in separate PR, but I don't have strong opinion.

The CI changes are tightly coupled as we are actually dropping the support for everything pre K8s-1.16

**Comment by bartsmykla** [comment]
You are right, sorry, I forgot about it!

**Pull request merged**